src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -112,11 +112,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                             '$(SelfContained)' != 'true'"
                  ResourceName="CompressionInSingleFileRequiresSelfContained" />
 
-    <!-- Enable warning for trying to use PublishRelease or PackRelease with a solution if env-var is not set.-->
-    <NETSdkWarning Condition="'$(PublishRelease)' != '' and '$(SolutionExt)' == '.sln' and '$(DOTNET_CLI_ENABLE_PUBLISH_RELEASE_FOR_SOLUTIONS)' == ''"
-                   ResourceName="PReleaseRequiresEnvVarOnSln" 
-                   FormatArguments="PublishRelease;DOTNET_CLI_ENABLE_PUBLISH_RELEASE_FOR_SOLUTIONS"
-    />
+    <!-- Projects in a solution cannot have conflicting configurations. This checks if PublishRelease conflicted at runtime to avoid extra project evaluations. -->
+    <NETSdkError  Condition="'$(_SolutionLevelPublishRelease)' != '' and '$(_SolutionLevelPublishRelease)' != '$(PublishRelease)'"
```

Comment:
Do we ever set `PublishRelease` (or `PackRelease`) in targets?  It doesn't look like it, so I think we will always generate these errors when publishing or packing a solution unless `PublishRelease` is specified, even if the solution only has a single project.

I think we may need to go back to setting the default values to these properties in the .targets files, so that we can generate the errors when there is a conflict but otherwise not generate them.

Reply:
They've been moved to targets, good idea. Not sure what I was thinking when I did it like the old implementation. This is no longer an issue, have tested it.